Content.Server/GameObjects/Components/Disposal/DisposalUnitComponent.cs

```diff
@@ -615,6 +646,14 @@ void IActivate.Activate(ActivateEventArgs eventArgs)
 
         async Task<bool> IInteractUsing.InteractUsing(InteractUsingEventArgs eventArgs)
         {
+            // you can dump trashbag content 
+            if (eventArgs.Using.HasTag("TrashBag"))
+            {
+                var dumpedTrashbag = TryEmptyTrashbag(eventArgs.User, eventArgs.Using);
+                if (dumpedTrashbag)
+                    return true;
+            }
```

Comment:
IMO better off making it a new component and dumping the logic onto a system given it'll make any future changes much easier e.g. if someone wanted to be able to empty one trash bag into another or something.